tools/lxdclient/client.go

```diff
@@ -262,6 +264,12 @@ func newRawClient(remote Remote) (*lxd.Client, error) {
 		return nil, errors.Trace(err)
 	}
 
+	// Replace the proxy handler with the one managed
+	// by Juju's worker/proxyupdater.
+	if tr, ok := client.Http.Transport.(*http.Transport); ok && tr.Proxy != nil {
```

Comment:
Yes, it does always do that. I've made it unconditional though, in case LXD changes that.